src/main/java/org/isf/medicalstock/manager/MovBrowserManager.java

```diff
@@ -122,4 +151,53 @@ private void check(LocalDateTime from, LocalDateTime to, String errMsgKey) throw
 			}
 		}
 	}
+
+	/**
+	 * Get the last Movement.
+	 *
+	 * @return the retrieved movement.
+	 * @throws OHServiceException
+	 */
+	public Movement getLastMovement() throws OHServiceException {
+		return ioOperations.getLastMovement();
+	}
+
+	/**
+	 * Deletes the last Movement.
+	 *
+	 * @param lastMovement - the last movement to delete
+	 * @throws OHServiceException
+	 */
+	@Transactional
+	public void deleteLastMovement(Movement lastMovement) throws OHServiceException {
+
+		MovementType movType = medicalDsrStockMovTypeManager.getMovementType(lastMovement.getType().getCode());
+		String lotCode = lastMovement.getLot().getCode();
+		Medical medical = lastMovement.getMedical();
+		int medicalCode = medical.getCode();
+		int quantity = lastMovement.getQuantity();
+		if (movType.getType().equals("+")) {
+			medical.setInqty(medical.getInqty() - quantity);
+			medicalsIoOperation.updateMedical(medical);
+			List<Integer> lastMovWithSameLot = ioOperations.getMedicalsFromLot(lotCode);
+			if (lastMovWithSameLot.size() == 1) {
+				lotRepository.deleteById(lotCode);
+			}
```

Comment:
If you want to be sure that the `Movement` you are deleting was creating a new lot from scratch, so it is safe to delete it, you may want to check if the same lot is used in other movements, so I think you can use the `findByLot` method for that.

Moreover the `getMedicalsFromLot` retrieves a list of `Medicals` from the lotCode, not `Movement`s. Since Lot<->Medical, its size will be always == 1. Indeed, I think that that method is an old code that was trying to retrieve a Medical from lots from movements: since `step_66_lot_information_at_the_wards_level.sql` we have Medical information in the Lot table itself with `LT_MDSR_ID`, so that method can be totally reworked and simplified.

Also, another method that can be safely deleted is `findAllByMedicalCodeOrderByLot_` because is identical to `findAllByMedicalCode` and never used.

If you want to adress also the last two problems above you are welcome, otherwise we can do it in another issue.

Thanks!